src/foam/u2/search/CurrencySearchView.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.search',
+  name: 'CurrencySearchView',
+  extends: 'foam.u2.View',
+
+  documentation: `
+    A SearchView for properties of type Currency. Lets the user filter by two
+    criteria:
+      1. A qualifier (Eg: equal to, not equal to, greater than)
+      2. An amount (Eg: 25.38)
+  `,
+
+  exports: [
+    'as data'
+  ],
+
+  requires: [
+    'foam.mlang.predicate.True',
+  ],
+
+  properties: [
+    {
+      name: 'property',
+      documentation: `The property that this view is filtering by. Should be of
+          type Currency.`,
+      required: true
+    },
+    {
+      class: 'String',
+      name: 'qualifier',
+      documentation: `Lets the user choose an MLang predicate to filter by.`,
+      view: {
+        class: 'foam.u2.view.ChoiceView',
+        choices: [
+          ['True', '--'],
+          ['Eq', 'is equal to'],
+          ['Neq', 'is not equal to'],
+          ['Gt', 'is greater than'],
+          ['Lt', 'is less than'],
+          ['Gte', 'is greater than or equal to'],
+          ['Lte', 'is less than or equal to'],
+        ],
+        defaultValue: 'True'
+      }
+    },
+    {
+      class: 'Float',
+      name: 'amount',
+      documentation: `The amount of money to filter by.`,
+      view: {
+        class: 'foam.u2.FloatView',
+        onKey: true,
+        precision: 2
+      }
+    },
+    {
+      name: 'predicate',
+      documentation: `All SearchViews must have a predicate as required by the
+          SearchManager. The SearchManager will read this predicate and use it
+          to filter the dao being displayed in the view.`,
+      expression: function(qualifier, amount) {
+        amount = typeof amount === 'number' ?
+            Math.floor(amount * 100) :
```

Comment:
If this is the only thing that distinguishes this from a regular number searchView, couldn't this whole class be made reusable and have just the line abstracted out to a template method in a very small sub-clas?

Reply:
Yeah, that makes sense. I'll split this into NumberSearchView and CurrencySearchView.